easybuild/easyblocks/m/metis.py

```diff
@@ -123,9 +123,22 @@ def sanity_check_step(self):
         if LooseVersion(self.version) < LooseVersion("5"):
             dirs += ["Lib"]
 
+        libext = []
+        if isinstance(self.cfg['configopts'], list):
+            for elements in self.cfg['configopts']:
+                if 'shared=1' in elements:
```

Comment:
this is a very involved bit of code, why not simply enable building of the `.so` by always including `shared=1`?

```python
# in configure_step
self.cfg.update('configopts', 'shared=1')
```

Alternatively, if you check for `shared=1` in `self.cfg['configopts']` in `configure_step`, you don't need to worry about checking whether `self.cfg['configopts']` is a list or not (you can assume it'll be a string when in `configure_step`, in fact, we already do when constructing `cmd`).